challenger/src/lib.rs

```diff
@@ -33,29 +39,47 @@ pub trait CanObserve<T> {
     }
 }
 
+/// A trait for sampling challenge elements from the Fiat-Shamir transcript.
+///
+/// Sampling produces pseudo-random elements deterministically derived
+/// from the absorbed inputs and the sponge state.
 pub trait CanSample<T> {
+    /// Sample a single challenge value from the transcript.
     fn sample(&mut self) -> T;
 
+    /// Sample an array of `N` challenge values from the transcript.
     fn sample_array<const N: usize>(&mut self) -> [T; N] {
         array::from_fn(|_| self.sample())
     }
 
+    /// Sample a `Vec` of `n` challenge values from the transcript.
     fn sample_vec(&mut self, n: usize) -> Vec<T> {
         (0..n).map(|_| self.sample()).collect()
     }
 }
 
+/// A trait for sampling random bitstrings from the Fiat-Shamir transcript.
 pub trait CanSampleBits<T> {
+    /// Sample a uniformly random `bits`-bit integer from the transcript.
+    ///
+    /// Guarantees that the returned value fits within the requested bit width.
     fn sample_bits(&mut self, bits: usize) -> T;
 }
 
+/// A high-level trait combining observation and sampling over a finite field.
 pub trait FieldChallenger<F: Field>:
     CanObserve<F> + CanSample<F> + CanSampleBits<usize> + Sync
 {
+    /// Absorb an algebraic element over the base field.
```

Comment:
`algebraic` isn't quite the right word here.
```suggestion
    /// Absorb an element from a vector space over the base field.
```